extension/app/ts/simulation/services/SimulationModeEthereumClientService.ts

```diff
@@ -98,9 +98,14 @@ export class SimulationModeEthereumClientService {
 		return min(baseFee + transaction.maxPriorityFeePerGas, transaction.maxFeePerGas)
 	}
 
-	public static mockSignTransaction = async (transaction: EthereumUnsignedTransaction) => {
-		const signatureParams = { r: 0n, s: 0n, yParity: 'even' as const }
+	public static mockSignTransaction = async (transaction: EthereumUnsignedTransaction) : Promise<EthereumSignedTransaction> => {
 		const unsignedTransaction = EthereumUnsignedTransactionToUnsignedTransaction(transaction)
+		if (transaction.type === 'legacy') {
+			const signatureParams = { r: 0n, s: 0n, v: 0n }
+			const hash = await keccak256.hash(serializeTransactionToBytes({ ...unsignedTransaction, ...signatureParams }))
+			return { ...transaction, ...signatureParams, hash }
+		}
+		const signatureParams = { r: 0n, s: 0n, yParity: 'even' as const }
 		const hash = await keccak256.hash(serializeTransactionToBytes({ ...unsignedTransaction, ...signatureParams }))
 		return { ...transaction, ...signatureParams, hash }
```

Comment:
Style: Consider an else here.  I like using if/else when both code paths are equally interesting/important.  In this case, `if (transaction.type === 'legacy'` isn't a guard clause but rather a primary code path, as is the other case.  This helps make it more clear that we are branch here, not guarding.
```suggestion
		if (transaction.type === 'legacy') {
			const signatureParams = { r: 0n, s: 0n, v: 0n }
			const hash = await keccak256.hash(serializeTransactionToBytes({ ...unsignedTransaction, ...signatureParams }))
			return { ...transaction, ...signatureParams, hash }
		} else {
			const signatureParams = { r: 0n, s: 0n, yParity: 'even' as const }
			const hash = await keccak256.hash(serializeTransactionToBytes({ ...unsignedTransaction, ...signatureParams }))
			return { ...transaction, ...signatureParams, hash }
		}
```